dmesg test flaky

Fails in circleci sometimes. For example: https://circleci.com/gh/u-root/u-root/27242

Error message:

--- FAIL: TestDmesg (0.08s)
    dmesg_test.go:17: Error: exit status 1, Output: 2019/03/29 23:53:13 syslog failed: operation not permitted
FAIL

--- a/cmds/dmesg/dmesg_test.go
+++ b/cmds/dmesg/dmesg_test.go
@@ -5,13 +5,17 @@
 package main
 
 import (
+	"os"
 	"testing"
 
 	"github.com/u-root/u-root/pkg/testutil"
 )
 
-// TODO(https://github.com/u-root/u-root/issues/1160): This test has been disabled
-func testDmesg(t *testing.T) {
+func TestDmesg(t *testing.T) {
+	if uid := os.Getuid(); uid != 0 {
+		t.Skipf("test requires root on CircleCI, your uid is %d", uid)
+	}
+
 	cmd := testutil.Command(t)
 	out, err := cmd.Output()
 	if err != nil || len(out) == 0 {

